### PR TITLE
feat(user): implement entity level authorization

### PR DESCRIPTION
**Pull request by racnan**
## Type of Change

- [ ] Bugfix
- [X] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
- Created a profile level user.
- Added entity level authorization in routes.

### Additional Changes

- [X] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
Dashboards users to be restricted at profile level.

## How did you test it?
Dashboard UI.

1. Login to dashboard.
2. Invite a user at profile level.
3. Login using profile level user.
4. Check if APIs are working.

## Checklist

- [X] I formatted the code `cargo +nightly fmt --all`
- [X] I addressed lints thrown by `cargo clippy`
- [X] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by ThisIsMani — DISMISSED** [review]

**Review by apoorvdixit88 — DISMISSED** [review]

File: crates/router/src/analytics.rs [hunk 1]
```diff
@@ -589,7 +605,10 @@ pub mod routes {
                     .await
                     .map(ApplicationResponse::Json)
             },
-            &auth::JWTAuth(Permission::Analytics),
+            &auth::JWTAuth {
+                permission: Permission::Analytics,
+                minimum_entity_level: EntityType::Merchant,
```
Comment (lsampras) [inline crates/router/src/analytics.rs:610]: This is profile level metrics

File: crates/router/src/analytics.rs [hunk 2]
```diff
@@ -472,7 +479,10 @@ pub mod routes {
...
```
Comment (lsampras) [inline crates/router/src/analytics.rs:484]: Profile level?

File: crates/router/src/analytics.rs [hunk 3]
```diff
@@ -429,7 +433,10 @@ pub mod routes {
...
```
Comment (lsampras) [inline crates/router/src/analytics.rs:438]: Org level

File: crates/router/src/analytics.rs [hunk 4]
```diff
@@ -546,7 +559,10 @@ pub mod routes {
...
```
Comment (lsampras) [inline crates/router/src/analytics.rs:564]: org

File: crates/router/src/analytics.rs [hunk 5]
```diff
@@ -663,7 +685,10 @@ pub mod routes {
...
```
Comment (lsampras) [inline crates/router/src/analytics.rs:690]: Org level

File: crates/router/src/analytics.rs [hunk 6]
```diff
@@ -706,7 +731,10 @@ pub mod routes {
...
```
Comment (lsampras) [inline crates/router/src/analytics.rs:736]: Profile level

**Review by lsampras — DISMISSED** [review]

**Review by ShankarSinghC — DISMISSED** [review]

**Review by ShankarSinghC — APPROVED** [review]

**Review by ThisIsMani — APPROVED** [review]

**Review by lsampras — APPROVED** [review]

**Pull request merged**